solr/core/src/java/org/apache/solr/handler/admin/api/ReplicationAPIBase.java

```diff
@@ -377,7 +383,9 @@ public void write(OutputStream out) throws IOException {
       try {
         initWrite();
 
-        Directory dir = solrCore.withSearcher(searcher -> searcher.getIndexReader().directory());
+        Directory dir =
+            filterDirectory(
+                solrCore.withSearcher(searcher -> searcher.getIndexReader().directory()));
```

Comment:
This seems to be the point of this PR.  Firstly, I'd rather see us get the Directory from the SolrCore.getDirectoryFactory as it doesn't seem very searcher-related.  Secondly, I think the DirectoryFactory ought to be able to return different different instances depending on the use-case/need.  See `DirectoryFactory.DirContext`.  Imagine having a new `REPLICATE` option that's only for moving bytes, for either backups or replication (both being similar)

Reply:
I like that proposal very much!
I'll close this PR and create another.

Reply:
New replacement [PR](https://github.com/apache/solr/pull/3185)